blocks/serializer.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * External dependencies
+ */
+import * as query from 'hpq';
+
+/**
+ * Internal dependencies
+ */
+import { getBlockSettings } from './registration';
+
+/**
+ * Takes a block list and returns the serialized post content
+ *
+ * @param  {Array}  blocks Block list
+ * @return {String}        The post content
+ */
+export default function serialize( blocks ) {
+	return blocks.reduce( ( memo, block ) => {
```

Comment:
As I noted above, it's hard to do without duplicating code (like computing rawContent and blockSettings several times) or passing those as parameters (which creates weird API)